### DISCUSSION TITLE
添加的文字层中的中文不显示？

### DISCUSSION THREAD

**Discussion by SilenSang** [Tencent/libpag#2178] [Q&A] [answered]
我向 PAG 文件中添加了一个文字层，显示出来发现，其中英文和数字显示正常，但中文没显示。代码段如下：
```
auto subtitleLayer = pag::PAGTextLayer::Make(5000000, "ABC123你好XYZ");
pagFile->addLayer(subtitleLayer);
```
显示结果：
![0](https://github.com/Tencent/libpag/assets/31496013/0cd54bda-4c6d-46e9-9da9-9796b5700743)
我已经设置不同的字体去测试，发现字体设置对英文有效，但中文依然不显示。可见不是字体的问题。
请问这是怎么回事？我应该怎么做？

**Comment by kevingpqi123** [comment]
字体注册请按照正规的方式来操作，可以参考下面的范例：
https://github.com/Tencent/libpag/blob/main/linux/src/main.cpp

**Reply by SilenSang** [reply]
跟字体无关。我可以正确设置字体的。
我怀疑跟字符编码有关。PAG 中的文字层对此有要求吗？

**Reply by kevingpqi123** [reply, marked as answer]
没有要求，这部分已经很成熟，服务端线上已经运行5、6 年了，如果你字体注册正确，中文不显示的原因可能是你选用的字体文件只包含英文符号和数字，没有包含中文，你找一个含有中文的字体试下

提一个建议：
反馈问题的时候将你的实现备注全，然后标注出不符合预期的地方，一次就可以定位到原因，从而提升对接效率
